src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -56,13 +53,12 @@ protected override bool Matches(string actual)
         }
 
         /// <summary>
-        /// Flag the constraint to use the supplied StringComparison type.
+        /// Overrides the <see cref="StringConstraint.IgnoreCase"/> setting and instead uses the specified comparison.
         /// </summary>
-        /// <param name="comparisonType ">The StringComparison type to be used.</param>
-        /// <returns>Self.</returns>
-        public SubstringConstraint Using(StringComparison comparisonType)
+        public SubstringConstraint Using(StringComparison? comparisonType)
         {
-            this.comparisonType = comparisonType;
+            this.comparisonType = comparisonType ?? (caseInsensitive ? 
+                StringComparison.CurrentCultureIgnoreCase : StringComparison.Ordinal);
```

Comment:
I did not know this till just now. `string.IndexOf` operates in `CurrentCulture` unless otherwise specified. I thought it was ordinal like `string.Equals` is. So @CharliePoole I was wrong. We should in fact have: `caseInsensitive ? StringComparison.CurrentCultureIgnoreCase : StringComparison.CurrentCulture`.

/cc @MhdTlb